### PR TITLE
[Github Gist] Optimise command

### PR DESCRIPTION
**Pull request by koinzhang**
## Description

- Simplify the code and improve performance
- Support saving gist as snippets

## Screencast

github-gist-1

github-gist-3

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by koinzhang** [comment]
@pernielsentikaer There are still some changes to be submitted

**Comment by koinzhang** [comment]
@pernielsentikaer Please keep it under review

File: extensions/github-gist/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   "license": "MIT",
   "preferences": [
     {
-      "name": "access-token",
+      "name": "personalAccessTokens",
```
Comment (pernielsentikaer) [inline extensions/github-gist/package.json:21]: Can we keep the old name?

File: extensions/github-gist/package.json [hunk 2]
```diff
@@ -75,6 +58,41 @@
               "value": "100"
             }
           ]
+        },
+        {
+          "name": "rememberTag",
```
Comment (pernielsentikaer) [inline extensions/github-gist/package.json:63]: Can we keep the old name?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by koinzhang** [comment]
@pernielsentikaer Please keep it under review

**Comment by mil3na** [comment]
Hi! Is this ready for review? If not, can you please mark it as a draft? Cheers!

**Comment by koinzhang** [comment]
@mil3na It's ready for review.

**Review by mil3na — APPROVED** [review]
Hi!
Thank you for your contribution! My name is Milena and I will be reviewing it today.

LGTM!

**Pull request merged**